src/lazy/expanded/compiler.rs

```diff
@@ -379,7 +413,7 @@ impl TemplateCompiler {
         context: EncodingContextRef<'_>,
         pending_macros: &'a MacroTable,
         macro_id: impl Into<MacroIdRef<'a>>,
-    ) -> Option<Rc<Macro>> {
+    ) -> Option<Arc<Macro>> {
```

Comment:
🗺️ Macros now reference each other with `Arc`, allowing the system `MacroTable` to be a lazily initialized singleton shared across all readers. The speedup offered by not having to allocate/copy the system table should outweigh the cost of creating `Arc`s.

There is no overhead when dereferencing an `Arc` (it's as fast as `&`), only when creating or destroying one.